File: data0/src/main/java/com/walhalla/mtprotolist/entity/MtprotoProxy.java
package com.walhalla.mtprotolist.entity;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.io.Serializable;

@Entity
public class MtprotoProxy implements Serializable {


    /* Not set Nullable */
    @PrimaryKey
    @ColumnInfo(name = "_id")
    public Long _id;

    @SerializedName("host")
    @Expose
    public String host = "";
    @SerializedName("port")
    @Expose
    public String port = "";
    @SerializedName("secret")
    @Expose
    public String secret = "";
    @SerializedName("code")
    @Expose
    public String code = "";
    @SerializedName("enabled")
    @Expose
    public Boolean enabled;

    public void setCity(String city) {
        this.city = city;
    }

    public void setRegionName(String regionName) {
        this.regionName = regionName;
    }

    public void setLat(double lat) {
        this.lat = lat;
    }

    public void setLon(double lon) {
        this.lon = lon;
    }

    @SerializedName("city")
    @Expose
    public String city;
    @SerializedName("regionName")
    @Expose
    public String regionName;
    @SerializedName("lat")
    @Expose
    public double lat;
    @SerializedName("lon")
    @Expose
    public double lon;

    @SerializedName("country")
    @Expose
    public String country;
    @SerializedName("region")
    @Expose
    public String region;
    @SerializedName("zip")
    @Expose
    public String zip;
    @SerializedName("timezone")
    @Expose
    public String timezone;
    @SerializedName("isp")
    @Expose
    public String isp;

    @SerializedName("org")
    @Expose
    public String org;

    @SerializedName("as")
    @Expose
    public String as;

    @Ignore
    public MtprotoProxy(String host, String port, String secret, String code, Boolean enabled) {
        this.host = host;
        this.port = port;
        this.secret = secret;
        this.code = code;
        this.enabled = enabled;
    }

    public MtprotoProxy() {
    } //FB


}

File: kmp/mtprotoshared/src/commonMain/kotlin/com/walhalla/mtprotolist/entity/MtprotoProxy.kt
package com.walhalla.mtprotolist.entity

import com.walhalla.mtproto.shared.config.AppConfig
import com.walhalla.mtproto.shared.util.HashUtils
import kotlinx.serialization.Serializable

@Serializable
class MtprotoProxy {
    var _id: Long? = null

    @JvmField
    var host: String? = ""

    @JvmField
    var port: String? = ""

    @JvmField
    var secret: String? = ""

    @JvmField
    var code: String? = ""

    var enabled: Boolean? = null

    var city: String? = null
    var regionName: String? = null
    var lat: Double = 0.0
    var lon: Double = 0.0
    var country: String? = null
    var region: String? = null
    var zip: String? = null
    var timezone: String? = null
    var isp: String? = null
    var org: String? = null
    var `as`: String? = null
    var update_at: Long = 0L

    constructor()

    constructor(host: String?, port: String?, secret: String?, code: String?, enabled: Boolean?) {
        this.host = host
        this.port = port
        this.secret = secret
        this.code = code
        this.enabled = enabled
    }

    fun shareUrl(): String = String.format(AppConfig.PROXY_HANDLER, host, port, secret)

    fun telegramUrl(): String = String.format(AppConfig.PROXY_HANDLER_TG, host, port, secret)

    fun firebaseKey(): String = HashUtils.md5(shareUrl())
}
